tests/tests_unit/test_cognite_client.py

```diff
@@ -102,6 +114,71 @@ def test_verify_ssl_enabled_by_default(self, rsps, client_config_w_token_factory
         assert client._api_client._http_client_with_retry.session.verify is True
         assert client._api_client._http_client.session.verify is True
 
+    def test_client_load(self):
+        config = {
+            "project": "test-project",
+            "client_name": "cognite-sdk-python",
+            "debug": True,
+            "credentials": {
+                "client_credentials": {
+                    "client_id": "test-client-id",
+                    "client_secret": "test-client-secret",
+                    "token_url": TOKEN_URL,
+                    "scopes": ["https://test.com/.default", "https://test.com/.admin"],
+                }
+            },
+        }
+        client = CogniteClient.load(config)
+        assert client.config.project == "test-project"
+        assert client.config.credentials.client_id == "test-client-id"
+        assert client.config.credentials.client_secret == "test-client-secret"
+        assert client.config.credentials.token_url == TOKEN_URL
+        assert client.config.credentials.scopes == ["https://test.com/.default", "https://test.com/.admin"]
+        assert client.config.debug is True
+        log = logging.getLogger("cognite.client")
+        log.handlers = []
+        log.propagate = False
+
+    def test_client_load_docs_example(self, set_env_vars):
+        """Test the example in the Quickstart docs, docs/source/quickstart.rst"""
```

Comment:
You may be able to use the doctests for this. Check out:
`tests/tests_unit/test_docstring_examples.py`

Reply:
That's what I was thinking, but it doesn't look like those tests cover anything in the `docs` folder?

Reply:
It finds the docstrings and runs the code examples (it finds), so it should be possible if you point it to the cognite client file.